src/codecs/webp/decoder.rs

```diff
@@ -321,16 +321,35 @@ impl<'a, R: 'a + Read> ImageDecoder<'a> for WebPDecoder<R> {
     }
 
     fn read_image(self, buf: &mut [u8]) -> ImageResult<()> {
-        assert_eq!(u64::try_from(buf.len()), Ok(self.total_bytes()));
```

Comment:
Let's not get rid of this assert. The spec for read_image [guarantees a panic](https://docs.rs/image/latest/image/trait.ImageDecoder.html#panics) if this condition doesn't hold.

Theoretically the values of `{vp8_frame, lossless_frame, extended}.get_buf_size()` should match, but there could be bugs